python/cudf/cudf/core/buffer/buffer.py

```diff
@@ -155,51 +139,53 @@ def __init__(
         self, data: Union[int, Any], *, size: int = None, owner: object = None
     ):
         if isinstance(data, int):
-            if size is None:
-                raise ValueError(
-                    "size must be specified when `data` is an integer"
-                )
-            if size < 0:
-                raise ValueError("size cannot be negative")
-            self._ptr = data
-            self._size = size
-            self._owner = owner
-        else:
-            if size is not None or owner is not None:
-                raise ValueError(
-                    "`size` and `owner` must be None when "
-                    "`data` is a buffer-like object"
-                )
+            warnings.warn(
```

Comment:
I have removed the warning. The idea was to simplify the code by only having `as_device_buffer_like()` handle integers. But now with the `ensure_buffer_like()`, I don't think it matters. 

> Is it because we can't construct a SpillableBuffer from a raw pointer and we need to align these?

No, `SpillableBuffer.__init__()` can just as well use `ensure_buffer_like()` to convert an integer to a buffer-like object.